components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -54,124 +67,527 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
 }
 
 - (void)commonBannerViewInit {
-  self.backgroundColor = [UIColor whiteColor];
+  self.backgroundColor = MDCColorFromRGB(MDCBannerDefaultBackgroundColor);
+  _containerView = [[UIView alloc] initWithFrame:CGRectZero];
+  _containerView.translatesAutoresizingMaskIntoConstraints = NO;
+  [self addSubview:_containerView];
 
-  _buttons = [[NSArray alloc] init];
-  self.numberOfButtons = 1;
+  _buttons = [[NSMutableArray alloc] init];
 
   UILabel *textLabel = [[UILabel alloc] init];
```

Comment:
Tip: Remove this and just use _ivar_ here.

Reply:
I used a local var here because I prefer to keep weak reference to subviews. In this case, ivar cannot be used to hold the reference before being added to its superview.

Reply:
What lead to the decision of a weak reference to subviews, as we typically hold a strong reference, just curious?

Reply:
I think it is more of a personal preference. I prefer not to put redundant strong reference in the class. I can change it to `strong` if it is a team convention.

Reply:
`strong` is not redundant in this case - it's an indication of ownership. Objects that own other objects should keep strong references to them. Objects that do not keep ownership can decide on strong/weak depending on the intended behavior.

Reply:
@romoore Superview always keeps a `strong` to the subview. And the `weak` means I need to to keep a reference to this subview so I can refer to it. If this is changed to `strong`, then there will be two strong pointers, which is redundant.

Reply:
You are right that superviews keep strong references to subviews here.  In fact, that's what Cody is suggesting. Since MDZBannerView "owns" that  UILabel, it should keep a strong reference to it.  It's not about redundancy or optimizing the number of strong references in the system, it's about clearly expressing ownership and retain semantics.  That UILabel is owned by the banner, so the banner's property should be strong to ensure ownership.  If I saw a UILabel property on a UIView, I would immediately think there was something odd about it and investigate.  At best it's a distraction, at worse it could cause programming errors later.

One potential problem is that a maintainer may add code to MDCBannerView removing the label from its superview (temporarily), assuming it follows the convention of strong retain semantics for ownership.  In that case, it could easily produce a bug (perhaps an inconsistent one) where the UILabel is sometimes released and other times when it's not.

If there are no remaining strong references to the MDCBannerView, then all of its retained objects (excepting some special cases) are also released. Unless there's  retain cycle or the semantic behavior of some property is such that it makes a `weak` pointer necessary, properties of owned objects should use `strong` semantics.

Reply:
Also, if anything is redundant, it's allocating more storage for the extra pointer required because of the `weak` semantics. If you use strong, you can save a pointer and just work with the iVar or property directly.

Reply:
Thanks for the thoughts, Rob. I agree with `weak` does occupy more resources and expressing ownership is a good practice.

For 
> One potential problem is that a maintainer may add code to MDCBannerView removing the label from its superview (temporarily), assuming it follows the convention of strong retain semantics for ownership. In that case, it could easily produce a bug (perhaps an inconsistent one) where the UILabel is sometimes released and other times when it's not.

this one, my thought is more aligned with the thought here: https://stackoverflow.com/questions/17517331/best-practice-for-custom-uiview-subview-strong-vs-weak, which is on the opposite side. I think your opinion is valid and can prevent some bugs caused by engineers removing subviews from parent unintentionally/intentionally. But I feel it is something the engineer who touch that part of code should be more aware of.

I wouldn't say I am fully convinced based on this point. But I can change it if it is a convention in our codebase. What do you think?

Reply:
Your StackOverflow link is talking about retain cycles where a view is keeping a strong link back to its superview. This is a very unusual situation, but I agree in that case `ownerView` should be a weak reference.  I believe in your code `UILabel` has no reference back to the Banner except the `superview` property, which is properly managed by lifecycle methods.    

Please make the label property use `strong` retain semantics - the banner owns the label.

Reply:
The question in StackOverflow was talking about subview reference superview, but I think the upvoted answer talked about the general ownership relationship which is larger than the scope of the question. Which I think it is valid in general cases.

The properties have been changed to `strong` in d24018618a336508639fa1127587190efc78086b since it is strongly recommended by the team.